lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -74,6 +75,11 @@ llvm::Error Provisioner::provision(DAGListTy &networks, Module &module) {
     }
   }
 
+  // Set collectConstants to false, this is because the DeviceManager will
+  // handle moving constants to the device, this way we can eliminate one
+  // copy operation.
+  cctx.backendOpts.collectConstants = false;
```

Comment:
It seems weird that a caller to `addNetwork()` could have set `backendOpts.collectConstants = true` and we would silently ignore it. Would it make sense instead to check that the caller set `collectConstants = false`, i.e. return an error if `!collectConstants`?

Reply:
I could see logging the difference, but don't like exiting with an error. collectConstants defaults to true but doesn't make sense in the runtime.

Reply:
Ok, yeah IMO it'd be a good idea to log something that says e.g. ignoring collectConstants.